stat/samplemv/halton.go

```diff
@@ -0,0 +1,166 @@
+package samplemv
+
+import (
+	"fmt"
+	"math/rand"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat/distmv"
+)
+
+// Haltoner is a wrapper around the Halton sampling generation method.
```

Comment:
But we have `Rejectioner`, `LatinHypercuber`, `MetropolisHastingser`. `Haltoner` is the name that fits that pattern.

Reply:
Those are unfortunate; they fit the naming convention for interfaces, which none of these is. I would name those `Rejection`, `LatinHypercube` and `MetropolisHastings`, but I think I have missed my opportunity when I should have spoken up (under that this would be `Halton`).

The comment wording should change though.

Reply:
Looking at the docs, I see why these are the way they are. Still unfortunate, and I guess this has to be this way.

Reply:
Let's put this in to match the pattern, and we can consider changing them en masse.

Reply:
In particular, maybe we just need the types and don't need the specific functions. The difference between  `LatinHypercube(batch,q, src)` and `LatinHypercube{Q: q, Src: src}.Sample(batch)` isn't so big.

Reply:
SGTM for further discussion.